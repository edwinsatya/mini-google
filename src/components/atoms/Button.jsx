const Button = ({ className, text = "btn", type = "button", onClick }) => {
  return (
    <button
      data-testid="button"
      type={type}
      className={`${className} cursor-pointer`}
      onClick={onClick}
    >
      {text}
    </button>
  );
};

export default Button;
