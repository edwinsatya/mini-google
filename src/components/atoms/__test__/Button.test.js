import { render, screen } from "@testing-library/react";
import Button from "../Button";
import renderer from "react-test-renderer";

describe("Render Button Atom Component", () => {
  it("Should be render without crashing", () => {
    render(<Button text="Search" />);
    const button = screen.getByTestId("button");
    expect(button).toBeTruthy();
  });

  it("Should be render text correctly", () => {
    render(<Button text="Search" />);
    const button = screen.getByTestId("button");
    expect(button).toHaveTextContent("Search");
  });

  it("Matches button snapshot", () => {
    const tree = renderer.create(<Button text="Search" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
