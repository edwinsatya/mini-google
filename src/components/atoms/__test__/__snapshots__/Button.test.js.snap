// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Atom Component Matches button snapshot 1`] = `
<button
  className="undefined cursor-pointer"
  data-testid="button"
  type="button"
>
  Search
</button>
`;
